### PR TITLE
Fix hrrr quilting restart tests

### LINKED ISSUE #1832: Several issues with how some of the hrrr 'quilt restart' tests are configured
There are several issues with how some of the hrrr 'quilt restart' tests are configured.

All quilt restart tests are supposed to exactly reproduce the restart outputs compared to the corresponding non-quilt restart tests, for example `hrrr_control_qr` must produce identical restart files as `hrrr_control`. In order to test this both of these tests have the same baseline directory (`CNTL_DIR` variable in test configuration file). Because of this only one of these two tests, and **only one** of these two tests, should be labeled as a test that generates new baselines, by having the string 'baseline' in the 4th column in rt.conf. This is not the case for these tests:

hrrr_control_qr  (intel version)
hrrr_control_qr_dyn32_phy32  (intel version)
hrrr_control_qr  (gnu version)
hrrr_control_qr_dyn32_phy32  (gnu version)

Another issue is that some 'qr' tests, I found one so far, have different baseline directory compared to the corresponding non-qr version. Tests `hrrr_control_dyn32_phy32` and `hrrr_control_qr_dyn32_phy32` should have the same baseline directory but they do not:

```
$ grep CNTL_DIR tests/hrrr_control_dyn32_phy32
export CNTL_DIR=hrrr_control_dyn32_phy32

$ grep CNTL_DIR tests/hrrr_control_qr_dyn32_phy32
export CNTL_DIR=hrrr_control_qr_dyn32_phy32
```

At some point we will switch to running all or most of the tests using quilting restart, but before we do that these issues must be fixed so that we are sure that quilting restart works correctly.

### PR DESCRIPTION
**Pull request by DusanJovic-NOAA**
## Description

Small change in quilting restart write routine to make all axis in the physics restart files use double values (instead of float even when physics is compiled in 32 bit) to make the files identical to FMS restart files. With this change we can have all tests that run quilting use the same baselines as corresponding control tests.

Is a change of answers expected from this PR?  No

### Issue(s) addressed

Link the issues to be closed with this PR, whether in this repository, or in another repository.
(Remember, issues should always be created before starting work on a PR branch!)
- fixes https://github.com/ufs-community/ufs-weather-model/issues/1832

## Testing

How were these changes tested?  ufs-weather-model regression test
What compilers / HPCs was it tested with?  Intel and GNU on Hera
Are the changes covered by regression tests? Yes
Have the ufs-weather-model regression test been run? Yes. On what platform?  Hera.
- Will the code updates change regression test baseline? If yes, why? Please show the baseline directory below.
- Please commit the regression test log files in your ufs-weather-model branch

## Dependencies

N/A

### CODE REVIEW

**Comment by BrianCurtis-NOAA** [comment]
Are there fixes that need to be made (or already are made) in FMS to accept float where applicable?

**Comment by DusanJovic-NOAA** [comment]
> Are there fixes that need to be made (or already are made) in FMS to accept float where applicable?

There are no changes required in FMS, this PR modifies fv3atm write routine to make the restart files identical to FMS restart files.

File: io/module_write_restart_netcdf.F90 [hunk 1]
```diff
@@ -565,7 +565,12 @@ subroutine write_out_ungridded_dim_atts_from_field(field, dimLabel, dimid, rc)
         ncerr = nf90_def_dim(ncid, trim(dimLabel), valueCount, dimid=dimid); NC_ERR_STOP(ncerr); NC_ERR_STOP(ncerr)
       endif
       if( typekind == ESMF_TYPEKIND_R4 ) then
-        ncerr = nf90_def_var(ncid, trim(dimLabel), NF90_FLOAT, dimids=(/dimid/), varid=varid); NC_ERR_STOP(ncerr)
+        !!! FIXME Use NF90_DOUBLE as axis type, even though axis data are float
+        !!!       This is needed to make phy/sfc restart files identical to FMS
+        !!!       restart files which always defines all axis as double
+
+        ! ncerr = nf90_def_var(ncid, trim(dimLabel), NF90_FLOAT, dimids=(/dimid/), varid=varid); NC_ERR_STOP(ncerr)
+        ncerr = nf90_def_var(ncid, trim(dimLabel), NF90_DOUBLE, dimids=(/dimid/), varid=varid); NC_ERR_STOP(ncerr)
```
Comment (junwang-noaa) [inline io/module_write_restart_netcdf.F90:573]: Is this a temporary solution? Will we allow the the axis type to be float real after the transition from FMS io? I feel that might be consistent with the axis data type defined in the model.

**Comment by DusanJovic-NOAA** [comment]
Currently, FMS restart files have all axis defined as double:
```
$ ncdump -h hrrr_control_dyn32_phy32_intel/RESTART/20210322.120000.phy_data.tile1.nc
netcdf \20210322.120000.phy_data.tile1 {
dimensions:
        xaxis_1 = 96 ;
        yaxis_1 = 96 ;
        zaxis_1 = 127 ;
        Time = UNLIMITED ; // (1 currently)
variables:
        double xaxis_1(xaxis_1) ;
                xaxis_1:cartesian_axis = "X" ;
        double yaxis_1(yaxis_1) ;
                yaxis_1:cartesian_axis = "Y" ;
        double zaxis_1(zaxis_1) ;
                zaxis_1:cartesian_axis = "Z" ;
        double Time(Time) ;
                Time:cartesian_axis = "T" ;
        float cv(Time, yaxis_1, xaxis_1) ;
                cv:checksum = "               0" ;
        float cvt(Time, yaxis_1, xaxis_1) ;
                cvt:checksum = "               0" ;
        float cvb(Time, yaxis_1, xaxis_1) ;
```
while quilting restart files have vertical axis defined as float:
```
$ ncdump -h hrrr_control_qr_dyn32_phy32_intel/RESTART/20210322.120000.phy_data.tile1.nc
netcdf \20210322.120000.phy_data.tile1 {
dimensions:
        xaxis_1 = 96 ;
        yaxis_1 = 96 ;
        zaxis_1 = 127 ;
        Time = UNLIMITED ; // (1 currently)
variables:
        double xaxis_1(xaxis_1) ;
                xaxis_1:cartesian_axis = "X" ;
        double yaxis_1(yaxis_1) ;
                yaxis_1:cartesian_axis = "Y" ;
        float zaxis_1(zaxis_1) ;
                zaxis_1:cartesian_axis = "Z" ;
        double Time(Time) ;
                Time:cartesian_axis = "T" ;
        float cv(Time, yaxis_1, xaxis_1) ;
                cv:checksum = "               0" ;
        float cvt(Time, yaxis_1, xaxis_1) ;
                cvt:checksum = "               0" ;
        float cvb(Time, yaxis_1, xaxis_1) ;
```
This PR fixes that inconsistency. Whether that should be changed, and whether we change it in both FMS and quilting is a separate issue. If we want to keep FMS and quilting restarts identical then the change should be made in both places at the same time.
The goal of this PR is to make quilting restart files identical to FMS, so that we can update all 'qr' tests to use same baselines as corresponding control tests.

**Review by BrianCurtis-NOAA — APPROVED** [review]

**Review by sadeghitabas — APPROVED** [review]

File: io/module_write_restart_netcdf.F90 (hunk 1 above)
Comment (zach1221) [reply io/module_write_restart_netcdf.F90:573]: @junwang-noaa has your question been satisfied here?

Comment (junwang-noaa) [reply io/module_write_restart_netcdf.F90:573]: @DusanJovic-NOAA can you answer this question?

Comment (DusanJovic-NOAA) [reply io/module_write_restart_netcdf.F90:573]: Yes. We will revert this change once FMS restart files are changed to output axis using the same type as data.

**Review by BrianCurtis-NOAA — APPROVED** [review]

**Comment by zach1221** [comment]
Combined PR #1467 is finished testing, and these fv3atm sub-PRs are ready for final review/merge. @junwang-noaa are you able to provide a review of this PR as well?

Edit disregard. changes here were combined into fv3atm 595

**Comment by DusanJovic-NOAA** [comment]
This PR is already merged via #595

**Pull request closed without merging**